### PR TITLE
ci: add Codecov integration

### PR DESCRIPTION
**Pull request by johnyob**
# Context

**Dependencies**: #536 
**Related Tasks**: [jstz-51](https://linear.app/tezos/issue/JSTZ-51/enable-code-coverage-in-ci)

Does what it says on the tin ☝️ 

# Description

# Manually testing the PR

### CODE REVIEW

File: package-lock.json
Comment (zcabter) [inline package-lock.json:1]: why did this file change?

**Comment by zcabter** [comment]
CI looks good, just 1 comment on an unexpected file change.

File: package-lock.json
Comment (johnyob) [reply package-lock.json:1]: Odd, I assume the change was added by my shell hook when running `direnv reload`. I'll remove it and see if it breaks anything 😅

Comment (johnyob) [reply package-lock.json:1]: Rebasing seems to have removed the artefact 🤷‍♂️

**Review by zcabter — APPROVED** [review]
Checked ci and the resultant coverage report - lgtm

**Pull request merged**